File: C/(2581)소수.c
#include<stdio.h>

void check_prime(int num, int div);

int main() {
	int num;
	scanf("%d", &num);
	check_prime(num, num-1);
}

void check_prime(int num, int div) {
	if (div > 1) {
		if (num % div == 0) printf("소수가 아님\n");
		else check_prime(num, div - 1);
	}
	else printf("소수\n");
}

File: BAEKJOON/Python/Math2/(2581)소수.py
import math

M = int(input())
N = int(input())

def check_prime(Num):
    if Num == 1:
        return 0
    # int() 로 바꿔주는 작업 중요
    n = int(math.sqrt(Num))
    
    while n >= 2:
        if Num % n == 0:
            return 0
        n -= 1
    return 1

first_prime = 0
sum = 0

for i in range(M,N+1):
    if check_prime(i) == 1:
        if sum == 0:
            first_prime = i
        sum += i

if sum == 0:
    print(-1)
else:
    print(sum)
    print(first_prime)